src/resources/env_config.rs

```diff
@@ -0,0 +1,233 @@
+        match self {
+            &BasicResource::UrlWithFallback{url: _, ref source} => source.clone(),
+            _ => panic!("no source available")
+        }
+    }
+}
+
+/// Special struct for font-awesome, as it consists of multiple embedded files
+/// but is configured as a directory. Methods implemented for this struct
+/// calculate the paths of all the files in this directory.
+pub struct Awesome {
+    resource: BasicResource
+}
+
+impl Resource for Awesome {
+    fn must_render_url(&self) -> bool {
+        self.resource.must_render_url()
+    }
+
+    fn must_embed(&self) -> bool {
+        self.resource.must_embed()
+    }
+
+    fn url(&self) -> String {
+        self.resource.url()
+    }
+
+    fn source(&self) -> Option<String> {
+        self.resource.source()
+    }
+}
+
+impl Awesome {
+    pub fn css_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/css/font-awesome.min.css")
+    }
+
+    pub fn eot_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.eot")
+    }
+
+    pub fn svg_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.svg")
+    }
+
+    pub fn ttf_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.ttf")
+    }
+
+    pub fn woff_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.woff")
+    }
+
+    pub fn woff2_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/fontawesome-webfont.woff2")
+    }
+
+    pub fn otf_source(&self) -> Option<String> {
+        self.resource.source().map(|p| p + "/fonts/FontAwesome.otf")
+    }
+}
+
+/// Configuration information for all the third-party resources. This specifies
+/// for every resource whether it should be included at all, and if so, just as
+/// an URL or also as an embedded resource.
+pub struct Configuration {
+    pub jquery: BasicResource,
+    pub mathjax: BasicResource,
+    pub highlight: BasicResource,
+    pub awesome: Awesome,
+    pub open_sans: BasicResource,
+    pub source_code_pro: BasicResource
+}
+
+/// Calculate an environment variable name of the format MDBOOK_[name]_[key]
+fn varname(resource : &str, key: &str) -> String {
+    "MDBOOK_".to_string() + &resource.to_uppercase() + "_" + key
+}
+
+/// Returns the contents of the environment variable of the given resource with
+/// the given key, or None if the environment variable was not set.
+fn var(resource: &str, key: &str) -> Option<String> {
+    env::var(varname(resource, key)).ok()
+}
+
+/// Returns a variable as var would. If the variable could not be found, return
+/// the given default.
+fn var_default(resource: &str, key: &str, default: &str) -> String {
+    var(resource, key).unwrap_or(String::from(default))
+}
+
+/// The rendering strategy for a resource
+enum Strategy {
```

Comment:
This could probably derive `Debug`, `Clone`, `Copy`, and `PartialEq` (the last one makes testing nicer).